File: traversal/bfs/javascript/BFS.js
class Graph{

    constructor(noOfVertices){
        this.V = noOfVertices
        this.edges = []
        this.visited = new Array(this.V).fill(false)
        this.result = []

        for(let i=0; i<this.V; i++)
            this.edges.push([]);
    }

    addEdge(src, dest){
        this.edges[src].push(dest);
    }

    bfsTraversal(s){
        //adding first node to queue
        let queue = [s]
        this.visited[s] = true

        while(queue.length!=0){
            let curr = queue.shift();

            for(let i=0; i<this.edges[curr].length; i++){
                let next = this.edges[curr][i];
                if(!this.visited[next]){
                    //adding adjacent nodes for current node
                    queue.push(next);
                    this.visited[next] = true;
                }
            }
            this.result.push(curr);
        }
        console.log(this.result);
    }

}

var graph = new Graph(11)

graph.addEdge(0, 1)
graph.addEdge(1, 4)
graph.addEdge(0, 2)
graph.addEdge(0, 3)
graph.addEdge(1, 5)
graph.addEdge(5, 7)
graph.addEdge(7, 9)
graph.addEdge(2, 6)
graph.addEdge(5, 8)
graph.addEdge(2, 5)
graph.addEdge(7, 10)
graph.addEdge(4, 7)
graph.addEdge(6, 8)

graph.bfsTraversal(0);

File: traversal/bfs/python/BFS.py
class Graph:

    def __init__(self, vertices):
        self.V = vertices
        self.graph = [[] for i in range(self.V)]

    def add_edge(self, src, dest):
        self.graph[src].append(dest)

    def bfs_traversal(self, s):
        visited = [False for i in range(self.V)]

        queue = [s]
        visited[s] = True

        while queue:
            curr = queue.pop(0)
            print(curr, end=' ')

            for adj in self.graph[curr]:
                if not visited[adj]:
                    queue.append(adj)
                    visited[adj] = True


graph = Graph(12)

graph.add_edge(1, 2)
graph.add_edge(2, 6)
graph.add_edge(6, 8)
graph.add_edge(1, 3)
graph.add_edge(8, 10)
graph.add_edge(1, 4)
graph.add_edge(3, 7)
graph.add_edge(2, 5)
graph.add_edge(6, 9)
graph.add_edge(3, 6)
graph.add_edge(8, 11)
graph.add_edge(5, 8)
graph.add_edge(7, 9)

graph.bfs_traversal(1)
